src/cluster.c

```diff
@@ -4410,6 +4410,43 @@ int getSlotOrReply(client *c, robj *o) {
     return (int) slot;
 }
 
+int getSlotStatus(client *c, unsigned char *slots, int del, int start_slot, int end_slot) {
```

Comment:
A weak preference I have is to validate the command is "syntactically correct" before doing any of the validation of the arguments. Today it looks like we don't do that, if you do `CLUSTER ADDSLOTS 1 FOO 2` it will throw an error if 1 is busy, even though the foo argument is invalid. The process then becomes:
1. Validate all the arguments and get the slots
2. Check to see if the slots are all valid
3. Apply the slots.

Reply:
@madolson, This is addressed in the latest commit. Please take a look. Thanks